ground/gcs/src/app/main.cpp

```diff
@@ -263,12 +263,17 @@ int main(int argc, char **argv)
     // Must be done before any QSettings class is created
     QSettings::setPath(XmlConfig::XmlSettingsFormat, QSettings::SystemScope,
             QCoreApplication::applicationDirPath()+QLatin1String(SHARE_PATH));
-    // keep this in sync with the MainWindow ctor in coreplugin/mainwindow.cpp
-    QSettings settings(XmlConfig::XmlSettingsFormat, QSettings::UserScope,
-                                 QLatin1String("TauLabs"), QLatin1String("TauLabs_config"));
 
-    overrideSettings(settings, argc, argv);
-    locale = settings.value("General/OverrideLanguage", locale).toString();
+    // Scope this so that we are guaranteed that the QSettings file is closed immediately. This
```

Comment:
> Huh. this really does that? there isn't a command like flush to make it happen immediately / more explicitly?

The closest command is ```sync()```. According to the docs:

> Writes any unsaved changes to permanent storage, and reloads any settings that have been changed in the meantime by another application.

>This function is called automatically from QSettings's destructor and by the event loop at regular intervals, so you normally don't need to call it yourself

Destroying it immediately prevents the helpful event loop from helpfully corrupting settings. The assumption is that it is this event loop, and not the destructor, that causes corruption when GCS segfaults.

Scoping forces the destructor to be called-- understood it's not guaranteed to be immediate-- and also removes the variable from local scope. Perhaps this might be seen as silly, but IMO it protects against future accidental use of the class. If some wayward soul decided to reuse ```settings``` outside the scope, it would cause a compiler failure instead of a runtime failure. This could also be accomplished with a function, but I'm not sure that's any cleaner.